mlkem/poly.h

```diff
@@ -81,7 +81,15 @@ __CPROVER_assigns(__CPROVER_object_whole(r));
 
 
 #define poly_decompress KYBER_NAMESPACE(poly_decompress)
-void poly_decompress(poly *r, const uint8_t a[KYBER_POLYCOMPRESSEDBYTES]);
+void poly_decompress(poly *r, const uint8_t a[KYBER_POLYCOMPRESSEDBYTES])
+/* *INDENT-OFF* */
+__CPROVER_requires(a != NULL)
+__CPROVER_requires(__CPROVER_is_fresh(a, KYBER_POLYCOMPRESSEDBYTES))
```

Comment:
What does `__CPROVER_is_fresh` mean?

Reply:
See https://diffblue.github.io/cbmc/contracts-memory-predicates.html
it basically means "This pointer has been allocated and is pointing at N bytes of memory"